__Задача 1__

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя уровень значимости 0.025 и правостороннюю критическую область, проверить данную гипотезу при условии, что в выборке из 100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.

__Решение:__

Нулевая гипотеза $H_0$: $a = 17$. Альтернативная гипотеза: $H_1$: $a > 17$. 

Найдем наблюдаемое значение:

$$T = \dfrac{\overline{X} - m}{\sigma_X}\sqrt{n} = \dfrac{17.5- 17}{2}\sqrt{100} = 2.5,$$

В предположении, что гипотеза $H_0$ верна, статистика $T$ имеет t-распределение $t(n - 1)$, где $n$ - объём выборки. Найдем критическую точку для односторонней критической области (при гипотезе $H_1$ : $a > 17$ ) по уровню значимости $\alpha$ = 0.025 : 

Нужный нам квантиль $t_{1 - \alpha, \: n - 1}$ найтем с помощью библиотеки scipy:

In [4]:
from scipy import stats
import numpy as np

In [5]:
alpha = 0.025
n = 100

In [6]:
stats.t.ppf(1 - alpha, n - 1)

1.9842169515086827

Поскольку наблюдаемые значения $T = 2.5 > 1.98$, то нулевую гипотезу о среднего диаметре шарика в 17 мм необходимо опровергнуть

__Задача 2__

Известно, что генеральная совокупность распределена нормально с известным средним квадратическим отклонением 16. Найти доверительный интервал для оценки математического ожидания `a` с надежностью 0.95, если выборочное среднее равно 80, а объём равен 256.

__Решение:__

Найдем доверительный интервал по формуле: 

$$
P \left( 
\overline{X} - t_{1 - \alpha/2, \: n - 1} \cdot \dfrac{\sigma_X}{\sqrt{n}} 
\leq M(X) \leq
\overline{X} + t_{1 - \alpha/2, \: n - 1} \cdot \dfrac{\sigma_X}{\sqrt{n}} 
\right) = 1 - \alpha.
$$

$\overline{X} = 80,  \sigma_X = 16, n = 256$

In [7]:
sig = 16
mean = 80 
n = 256

In [8]:
(mean - 2*sig/np.sqrt(n), mean + 2*sig/np.sqrt(n))

(78.0, 82.0)

__Задача 3__

Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
```
202, 203, 199, 197, 195, 201, 200, 204, 194, 190
```
Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что уровень значимости равен 1%?

__Решение:__

Найдем доверительный для проверки гипотезы, утверждающей, что средний вес пачки печения составляет 200г:

In [11]:
sample = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

In [13]:
m = sample.mean()

m

198.5

In [18]:
n = 10

In [19]:
sig = np.sqrt(((sample - m)**2).sum()/(n-1))

sig

4.453463071962462

In [28]:
alpha = 0.01

Используем библиотеку stats для нахождения t квантиля: 

In [30]:
t_3 = stats.t.ppf(1 - alpha, n - 1)

t_3 

2.8214379233005493

Отсюда, доверительный интвервал: 

In [32]:
(m - t_3*sig/np.sqrt(n), m + t_3*sig/np.sqrt(n))

(194.52654448737295, 202.47345551262705)

Утверждение продавца верно.